src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt

```diff
@@ -15,34 +19,29 @@ class AuthenticationFilter : ContainerRequestFilter {
     companion object {
         const val cookieName = "Jwt"
         const val ignoreThisUrl = "api/login"
-        const val employeeIdHeader = "X-Employee-Id"
-        const val companyIdHeader = "X-Company-Id"    }
+    }
 
     override fun filter(requestContext: ContainerRequestContext) {
 
         if (requestContext.uriInfo.path == ignoreThisUrl) return
 
         val jwtCookie = requestContext.cookies[cookieName]?.value
         val claims = verifyAndExtractClaims(jwtCookie, requestContext)
-        claims?.let{
-            //TO DO: create a User object here and store that as a request attribute
-            requestContext.headers.add(employeeIdHeader, it["employeeId"].toString())
-            requestContext.headers.add(companyIdHeader, it["companyId"].toString())
+        claims?.let {
+            val activeUser =
+                ActiveUser(UUID.fromString(it["employeeId"].toString()), UUID.fromString(it["companyId"].toString()), Role.valueOf(it["role"].toString().uppercase()), Department.valueOf(it["department"].toString().uppercase()))
+            requestContext.setProperty("activeUser", activeUser)
```

Comment:
you already did the toString in the verify method, no?

Reply:
if im removing the toString I need to use !!.uppercase()
beacuse the toString will assert me that it wont be null

Reply:
I think its ok to use it here because the claims are Map<String, String>?
and the second string is not nullable anymore

Reply:
verifyAndExtractClaims should return a Map<String,String>, not nullable

Reply:
Its already does in local and still i get:
e: file:///Users/boaz.cohen/workspace/hibob-academy/src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt:32:142 **Only safe (?.) or non-null asserted (!!.) calls are allowed on a nullable receiver of type String?**